src/library_webgpu.js

```diff
@@ -703,32 +714,39 @@ var LibraryWebGPU = {
       });
     }
     deviceWrapper.lostCallback = function(info) {
-      var messagePtr = allocateUTF8(info.message);
-      {{{ makeDynCall('viii', 'callback') }}}(
-          WebGPU.DeviceLostReason[info.reason], messagePtr, userdata);
-      _free(messagePtr);
+      // This will skip the callback if the runtime is no longer alive.
```

Comment:
callUserCallback does this:

```
    if (ABORT) {                                                                    
#if ASSERTIONS                                                                      
      err('user callback triggered after application aborted.  Ignoring.');         
#endif                                                                              
      return;                                                                       
    }      
```

I'm not sure if that is exactly the same is checking the runtime is alive... but in any case if there is any checking to be done it should happen in `callUserCallback` itself and not here.

Reply:
Yes, I think we could add an assert for that (since if that happens I think it would be programmer error).   Separate PR for that makes sense I think.

Reply:
Draft PR was: https://github.com/emscripten-core/emscripten/pull/15256

Reply:
If we exit the runtime shouldn't unregistered any callbacks we have with the browser.  Is that possible?

Reply:
It is possible in principle for the `onuncapturederror` since that's an event handler (if we have a place to do work on exiting the runtime), but it's not possible for DeviceLost, here, because promise.then isn't cancellable. We can only cancel it by checking something (like `runtimeExited`) inside the .then callback.